### PR TITLE
speed.c: simplify aggregation of -multi results

### PR DESCRIPTION
**Pull request by cristian-stoica**

### CODE REVIEW

**Comment by stevelinsell** [comment]
This change is very good. We were seeing problems with speed on faster boards where the accuracy of the measurements was impaired due to the value sent down the pipe being rounded. With this change (fix) the rounding only happens at the final step so the accuracy is maintained. It's a shame this fix is only flagged for post 1.1.0 as it does not look very risky or controversial.

File: apps/speed.c [hunk 1]
```diff
@@ -2688,8 +2688,8 @@ int speed_main(int argc, char **argv)
                    k, rsa_bits[k], rsa_results[k][0], rsa_results[k][1]);
         else
             printf("rsa %4u bits %8.6fs %8.6fs %8.1f %8.1f\n",
```
Comment (stevelinsell) [inline apps/speed.c]: Now you have changed the maths round, do we still require 6 decimal places on these fields? I think the info in the decimal places will now get thrown away after the maths on the other end of the pipe when the data is formatted for output. This also applies to DSA, ECDH, ECDSA as well.

Comment (cristian-stoica) [inline apps/speed.c]: Thanks for review.
I guess the extended precision doesn't matter any more. All other things being equal, for the purpose of the patch I usually go with minimum changes to ease up reviewer's work.

**Comment by stevelinsell** [comment]
BUMP
Now Master branch is open for post 1.1.0 changes, is there any chance this fix can be reviewed and merged by the core team?
The title is slightly misleading as this pull request is not just a simplification but actually fixes a rounding bug which causes measurements to be very inaccurate when using openssl speed with very fast hardware engines.
In my opinion it is a bug fix and should be backported into 1.1.0 as well, but that's just my opinion :-)

**Comment by levitte** [comment]
BUMP.  This branch needs to be rebased on a fresher master.  @cristian-stoica, are you still with us?

**Comment by richsalz** [comment]
CLA.

**Comment by cristian-stoica** [comment]
Yes I am - From my point of view this is a trivial patch. So you can merge it as such.

Regarding CLA, all I can say is that the issue was discussed again about a week ago without a clear resolution schedule...

**Comment by levitte** [comment]
@richsalz, I can agree with @cristian-stoica that this is a trivial patch (or rather, four trivial patches). Since you labeled this with need-cla, maybe you have a different opinion?

**Comment by richsalz** [comment]
If you are both satisfied, that's okay.  Recall that each commit should have a "CLA: trivial" line in it.

**Comment by levitte** [comment]
Ok...

@cristian-stoica, unfortunately, it seems that `apps/speed.c` has moved on and there are conflicts.  I did give it a go just now, thinking that the conflict might be trivial as well...  but alas, one of the conflicts is a little too much to handle for me (around `ecdsa_results`), so if you could have a look and update this PR, that would be grand.

**Comment by cristian-stoica** [comment]
Sure. I'll have a look at it

**Comment by cristian-stoica** [comment]
@levitte kdiff3 was a big help and handled the rebase. The conflicting lines came from removing of an #ifdef in d6073e27e (Small nits and cleanups).  The other conflicts were from whitespaces and newlines changes.

**Comment by richsalz** [comment]
Please push an update that adds "CLA: trivial" to the commits (better if original poster does it, not us)
+1 assuming that's done.

**Comment by mattcaswell** [comment]
Ping @levitte for merge.

**Comment by levitte** [comment]
Merged, master only.

**Pull request closed without merging**